libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -630,6 +630,104 @@ string YulUtilFunctions::overflowCheckedIntExpFunction(
 	});
 }
 
+string YulUtilFunctions::overflowCheckedIntLiteralExpFunction(
+	RationalNumberType const& _baseType,
+	IntegerType const& _exponentType,
+	IntegerType const& _commonType
+)
+{
+	solAssert(!_exponentType.isSigned(), "");
+	solAssert(_baseType.isNegative() == _commonType.isSigned(), "");
+	solAssert(_commonType.numBits() == 256, "");
+
+	// Calculates the upperbound for exponentiation, that is, calculate `b`, such that
+	// _base**b <= _maxValue and _base**(b + 1) > _maxValue
+	auto findExponentUpperbound = [](bigint const _base, bigint const _maxValue) -> unsigned
+	{
+		unsigned first = 0;
+		unsigned last = 255;
+		unsigned middle;
+
+		while (first < last)
+		{
+			middle = (first + last) / 2;
+
+			if (
+				// The condition on msb is a shortcut that avoids computing large powers in
+				// arbitrary precision.
+				boost::multiprecision::msb(_base) * middle <= boost::multiprecision::msb(_maxValue) &&
+				boost::multiprecision::pow(_base, middle) <= _maxValue
+			)
+			{
+				if (boost::multiprecision::pow(_base, middle + 1) > _maxValue)
+					return middle;
+				else
+					first = middle + 1;
+			}
+			else
+				last = middle;
+		}
+
+		return last;
```

Comment:
Can you add a solAssert that verifies `_base**b <= _maxValue and _base**(b + 1) > _maxValue`?

Reply:
Aren't there asserts like that later on?

Reply:
Ah right!